components/iotserver-mgt/org.wso2.carbon.device.mgt.iot.common/src/main/java/org/wso2/carbon/device/mgt/iot/common/transport/mqtt/MqttPublisher.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.device.mgt.iot.common.transport.mqtt;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.eclipse.paho.client.mqttv3.*;
+import org.wso2.carbon.device.mgt.iot.common.controlqueue.ControlQueueConnector;
+import org.wso2.carbon.device.mgt.iot.common.exception.DeviceControllerException;
+
+import java.io.File;
+import java.util.HashMap;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * The Class MqttControlPublisher. It is an implementation of the interface
+ * ControlQueueConnector.
+ * This implementation supports publishing of control signals received to an
+ * MQTT end-point.
+ * The configuration settings for the MQTT end-point are read from the
+ * 'devicecloud-config.xml' file of the project.
+ * This is done using the class 'DeviceCloudConfigManager.java' which loads
+ * the settings from the default xml org.wso2.carbon.device.mgt.iot.common.devicecloud.org.wso2.carbon.device.mgt.iot.common.config.server.configs
+ * file -
+ * /resources/conf/device-controls/devicecloud-config.xml
+ */
+public class MqttPublisher implements MqttCallback {
+
+	private static final Log log = LogFactory.getLog(MqttPublisher.class);
+
+	private String mqttEndpoint;
+	private String mqttUsername;
+	private String mqttPassword;
+	private boolean mqttEnabled = false;
+
+	public MqttPublisher() {
+	}
+
+	public void initControlQueue() throws DeviceControllerException {
+		mqttEndpoint = MqttConfig.getInstance().getMqttQueueEndpoint();
+		mqttUsername = MqttConfig.getInstance().getMqttQueueUsername();
+		mqttPassword = MqttConfig.getInstance().getMqttQueuePassword();
+		mqttEnabled = MqttConfig.getInstance().isEnabled();
+	}
+
+	public void publish(String publishClientId, String publishTopic, byte[] payload)
+			throws DeviceControllerException {
+
```

Comment:
MQTT transport has been moved to input & output adapters in [cdmf-device-mgt-plugins](https://github.com/wso2/carbon-device-mgt-plugins/tree/master/components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.input.adapter.mqtt/src/main/java/org/wso2/carbon/device/mgt/iot/input/adapter/mqtt/util). This class does not exist anymore.